src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -202,12 +211,15 @@ public BitArray(BitArray bits)
         =========================================================================*/
         public bool Get(int index)
         {
-            if (index < 0 || index >= Length)
+            if ((uint)index >= (uint)Length)
             {
                 throw new ArgumentOutOfRangeException(nameof(index), index, SR.ArgumentOutOfRange_Index);
             }
 
-            return (m_array[index / 32] & (1 << (index % 32))) != 0;
+            int quotient = index >> BitShiftPerInt32;   // Divide by 32.
+            int remainder = index - (quotient << BitShiftPerInt32);   // Multiply by 32.
```

Comment:
Everyone writes this as `index & 31`. Or, if you want to be pedantic, `((1 << BitShiftPerInt32) - 1)`. But IMO this particular use of constants does more harm than good. The old code was perfectly clear to anyone who knows what a bit array is and its only problem is that the JIT isn't good enough to figure out that index is always positive.

And while I'm at it - `quotient` and `remainder` are poor names in this context. I'd expect something more like `elementIndex` and `bitIndex`.